### PR TITLE
feat(LIB-434): checkbox and checkbox group

### PR DESCRIPTION
**Pull request by Crish15**
Jira: [LIB-434](https://fiscozen.atlassian.net/browse/LIB-434)

### CODE REVIEW

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  "flex flex-col justify-center",
+  props.size === "sm"
+    ? "gap-6 checkbox-group--child-container__small"
+    : "gap-8 checkbox-group--child-container__medium",
+]);
+</script>
+<style scoped>
+.checkbox-group--child-container__small {
+  margin-top: 5px;
+  padding-left: 24px;
```
Comment (oxcened) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:99]: there's pl-24

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+.checkbox-group--child-container__medium {
+  margin-top: 6px;
+  padding-left: 24px;
```
Comment (oxcened) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:104]: same here

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  } else {
+    // remove parent value from model if it exists
+    if (model.value.includes(props.value || props.label))
+      model.value = model.value.filter(
+        (value) => value !== (props.value || props.label),
+      );
+  }
+}
+
+function onCheckboxParentChange() {
```
Comment (oxcened) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:69]:
```suggestion
function handleCheckboxParentChange() {
```

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    :value="props.value"
+    :label="props.label"
+    :disabled="disabled"
+    :emphasis="emphasis"
+    :error="error"
+    :size="size"
+    :indeterminate="isIndeterminate"
+    @change="onCheckboxParentChange"
+  >
+    <template #children v-if="children && children.length > 0">
```
Comment (oxcened) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:13]:
```suggestion
    <template #children v-if="children?.length">
```

File: packages/checkbox/src/FzCheckbox.vue [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+const checkValueIsInModel = () => {
+  if (typeof model.value === "boolean") {
+    return model.value;
+  } else {
+    return model.value.includes(value);
+  }
+};
+
+const onChange = (e: any) => {
```
Comment (oxcened) [inline packages/checkbox/src/FzCheckbox.vue:109]: per una riga, si potrebbe fare inline

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+]);
+</script>
+<style scoped>
+.checkbox-group--child-container__small {
+  margin-top: 5px;
+  padding-left: 24px;
+}
+
+.checkbox-group--child-container__medium {
+  margin-top: 6px;
```
Comment (oxcened) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:103]: there's mt-6

File: packages/checkbox/src/components/FzCheckboxGroupOption.vue [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+const model = defineModel<string[]>({
+  required: true,
+  default: [],
+});
+
+const staticChildContainerClass = "flex flex-col justify-center pl-24";
+
+const computedChildContainerClasses = computed(() => [
+  props.size === "sm" ? "gap-6 mt-[5px]" : "",
+  props.size === "md" ? "gap-8 mt-[6px]" : "",
```
Comment (Crish15) [inline packages/checkbox/src/components/FzCheckboxGroupOption.vue:46]: change to mt-6

**Pull request merged**